src/typing_extensions.py

```diff
@@ -3524,16 +3526,47 @@ def _raise_attribute_error(self, name: str) -> Never:
         def __repr__(self) -> str:
             return self.__name__
 
-        def __getitem__(self, parameters):
-            if not isinstance(parameters, tuple):
-                parameters = (parameters,)
-            parameters = [
-                typing._type_check(
-                    item, f'Subscripting {self.__name__} requires a type.'
-                )
-                for item in parameters
-            ]
-            return typing._GenericAlias(self, tuple(parameters))
+        if sys.version_info >= (3, 11):
+            def __getitem__(self, parameters):
+                if not isinstance(parameters, tuple):
+                    parameters = (parameters,)
+                parameters = [
+                        typing._type_check(
+                            item, f'Subscripting {self.__name__} requires a type.'
+                        )
+                        for item in parameters
+                    ]
+                alias = typing._GenericAlias(self, tuple(parameters))
```

Comment:
The typing variant sets the `__value__` and `__type_params__` on the Generic. Also adding this here. Possibly move this to another PR.

Reply:
> The typing variant sets the __value__ and __type_params__ on the Generic

Not exactly; instead it creates a `types.GenericAlias`, which proxies these attributes. I believe we could also use `types.GenericAlias` here (though not on 3.8 where it doesn't exist).

Reply:
However, this PR has a better chance of being merged if you limit it to fixing one issue at a time. Feel free to open additional PRs or issues for other improvements.

Reply:
> > The typing variant sets the **value** and **type_params** on the Generic
> 
> Not exactly; instead it creates a `types.GenericAlias`, which proxies these attributes. I believe we could also use `types.GenericAlias` here (though not on 3.8 where it doesn't exist).

Thanks I'll try to strip out the points where its easily possible.

I've now created the appropriate issue for this.
- #469

Reply:
> > The typing variant sets the **value** and **type_params** on the Generic
> 
> Not exactly; instead it creates a `types.GenericAlias`, which proxies these attributes. I believe we could also use `types.GenericAlias` here (though not on 3.8 where it doesn't exist).

I think this works great, I applied this now to this PR as well and it cleans up some tests. Needed some rework in __getitem__ again.